catalogs/catalog-jdbc-clickhouse/src/main/java/org/apache/gravitino/catalog/clickhouse/converter/ClickHouseTypeConverter.java

```diff
@@ -0,0 +1,193 @@
+  static final String FIXEDSTRING = "FixedString";
+  static final String DATE = "Date";
+  static final String DATE32 = "Date32";
+  static final String DATETIME = "DateTime";
+  static final String DATETIME64 = "DateTime64";
+  static final String ENUM = "Enum";
+  static final String BOOL = "Bool";
+  static final String UUID = "UUID";
+
+  // bellow is Object Data Type
+  static final String IPV4 = "IPv4";
+  static final String IPV6 = "IPv6";
+  static final String ARRAY = "Array";
+  static final String TUPLE = "Tuple";
+  static final String MAP = "Map";
+  static final String VARIANT = "Variant";
+  static final String LOWCARDINALITY = "LowCardinality";
+  static final String NULLABLE = "Nullable";
+  static final String AGGREGATEFUNCTION = "AggregateFunction";
+  static final String SIMPLEAGGREGATEFUNCTION = "SimpleAggregateFunction";
+  static final String GEO = "Geo";
+
+  // bellow is Special Data Types
+  static final String Domains = "Domains";
+  static final String Nested = "Nested";
+  static final String Dynamic = "Dynamic";
+  static final String JSON = "JSON";
+
+  @Override
+  public Type toGravitino(JdbcTypeBean typeBean) {
+    String typeName = typeBean.getTypeName();
+    if (typeName.startsWith("Nullable(")) {
+      typeName = typeName.substring(9, typeName.length() - 1);
+    }
+
+    if (typeName.startsWith("Decimal(")) {
+      typeName = "Decimal";
+    }
+
+    if (typeName.startsWith("FixedString(")) {
+      typeName = "FixedString";
+    }
+
+    switch (typeName) {
+      case INT8:
+        return Types.ByteType.get();
+      case INT16:
+        return Types.ShortType.get();
+      case INT32:
+        return Types.IntegerType.get();
+      case INT64:
+        return Types.LongType.get();
+      case UINT8:
+        return Types.ByteType.unsigned();
+      case UINT16:
+        return Types.ShortType.unsigned();
+      case UINT32:
+        return Types.IntegerType.unsigned();
+      case UINT64:
+        return Types.LongType.unsigned();
+      case FLOAT32:
+        return Types.FloatType.get();
+      case FLOAT64:
+        return Types.DoubleType.get();
+      case DECIMAL:
+        return Types.DecimalType.of(typeBean.getColumnSize(), typeBean.getScale());
+      case STRING:
+        return Types.StringType.get();
+      case FIXEDSTRING:
+        return Types.FixedCharType.of(typeBean.getColumnSize());
+      case DATE:
+        return Types.DateType.get();
+      case DATE32:
+        return Types.DateType.get();
+      case DATETIME:
+        return Types.TimestampType.withoutTimeZone();
+      case DATETIME64:
+        return Types.TimestampType.withoutTimeZone();
+      case BOOL:
+        return Types.BooleanType.get();
+      case UUID:
+        return Types.UUIDType.get();
+      default:
+        return Types.ExternalType.of(typeBean.getTypeName());
+    }
+  }
+
+  @Override
+  public String fromGravitino(Type type) {
+    if (type instanceof Types.ByteType) {
```

Comment:
> I'd do `typeName = type.getTypeName()` and use a `switch` statement for the name. The `if ... else if` is a headache to me.

MySql, Doris,OceanBase, psql ，it's all like this. Perhaps maintaining this would be better？Should we use one issue to make the changes uniformly in the future.

Reply:
> My gut feeling is that this conversion between type and type names can be generalized and then put to `JdbcTypeConverter`. We can then maintain the peculiarities for ClickHouse here.


I thinl, in fact, it is simply taking its own type from the gravitino type, processing mostly the types of its own engine, most of which are different. Unless the type of gravitino is separated and a callback is written to return its own type. But the code is also quite obscure.

Reply:
My suggestion was to promote this to the JDBC level from ClickHouse level.
I was not suggesting to promote this to the Gravitino level.